### PR TITLE
Feature/main constructor

### PR DESCRIPTION
**Pull request by Den313**
Added main constructor functionality. This feature allows moderators to create containers, where they can customize main page view by adding, deleting sections, their names and entitites, placed in this sections, choose if recently played (Your mix) section(s) needed to show or not. System can have many containers, but only one of them can be published and result of publishing will be shown in main page.

### CODE REVIEW

File: frontend/src/app/modules/main/cards/constructor-container-card/constructor-container-card/constructor-container-card.component.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  clickEmiter = new EventEmitter<void>();
+
+  constructor(
+    private _constructorService: ConstructorService,
+    private _router: Router,
+    private _snackbarService: SnackbarService
+  ) { }
+
+  publishContainer = () => {
+    this._constructorService.publishContainer(this.container).subscribe();
```
Comment (sleepergoose) [inline frontend/src/app/modules/main/cards/constructor-container-card/constructor-container-card/constructor-container-card.component.ts]: There is a subscription without unsubscribing. Use takeUntil, take and so on `this._constructorService.publishContainer(this.container).subscribe()`

File: frontend/src/app/modules/main/cards/constructor-container-card/constructor-container-card/constructor-container-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    this._router.navigateByUrl('/home');
+  };
+
+  deleteContainer = () => {
+    this._constructorService.deleteContainer(this.container.id).subscribe(
```
Comment (sleepergoose) [inline frontend/src/app/modules/main/cards/constructor-container-card/constructor-container-card/constructor-container-card.component.ts]: Also there is a subscription without unsubscribing. 
`this._constructorService.deleteContainer(this.container.id).subscribe`

**Review by sleepergoose — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/main/main-home/main-home.component.ts [hunk 3]
```diff
@@ -60,6 +68,13 @@ export class MainHomeComponent implements OnInit, OnDestroy {
   }
 
   async ngOnInit() {
+    this._constructorService.getPublishedContainer().subscribe(
```
Comment (sleepergoose) [inline frontend/src/app/modules/main/main-home/main-home.component.ts]: Unsubscribing needs. Something like `.pipe(takeUntil(this._unsubscribe$))`

File: frontend/src/app/modules/perflow-studio/constructor/containers-page/containers-page/containers-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  containers: ContainerView[] = [] as ContainerView[];
+
+  constructor(
+    private _constructorService: ConstructorService
+  ) {
+
+  }
+
+  public ngOnInit() {
+    this._constructorService.getAllContainersViews().subscribe(
```
Comment (sleepergoose) [inline frontend/src/app/modules/perflow-studio/constructor/containers-page/containers-page/containers-page.component.ts]: There is a subscription without unsubscribing.

File: frontend/src/app/modules/perflow-studio/constructor/create-edit-container/create-edit-container/create-edit-container.component.ts [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    }
+  }
+
+  public ngOnDestroy() {
+    this._unsubscribe$.next();
+    this._unsubscribe$.complete();
+  }
+
+  startEditMode = () => {
+    this._constructorService.getContainer(this._id!).subscribe(
```
Comment (sleepergoose) [inline frontend/src/app/modules/perflow-studio/constructor/create-edit-container/create-edit-container/create-edit-container.component.ts]: There is a subscription without unsubscribing.

File: frontend/src/app/modules/perflow-studio/constructor/create-edit-container/create-edit-container/create-edit-container.component.ts [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+    };
+  }
+
+  saveName = () => {
+    this.editNamePosition = -1;
+  };
+
+  saveContainer = () => {
+    if (this._id) {
+      this._constructorService.updateContainer(this.container).subscribe(
```
Comment (sleepergoose) [inline frontend/src/app/modules/perflow-studio/constructor/create-edit-container/create-edit-container/create-edit-container.component.ts]: There is a subscription without unsubscribing.

File: frontend/src/app/modules/perflow-studio/constructor/create-edit-container/create-edit-container/create-edit-container.component.ts [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+          this._snackbarService.show({
+            message: 'Your changes confirmed.',
+            header: 'Container edited successfully!'
+          });
+          this._router.navigateByUrl('/perflowstudio/constructor');
+        }
+      );
+    }
+    else {
+      this._constructorService.createContainer(this.container).subscribe(
```
Comment (sleepergoose) [inline frontend/src/app/modules/perflow-studio/constructor/create-edit-container/create-edit-container/create-edit-container.component.ts]: And here

File: frontend/src/app/modules/perflow-studio/perflow-studio-routing.module.ts [hunk 8]
```diff
@@ -15,7 +18,7 @@ const routes: Routes = [{
   path: '',
   component: MainPageComponent,
   canActivate: [PerflowStudioGuard],
-  canActivateChild: [PerflowStudioGuard],
+  // canActivateChild: [PerflowStudioGuard],
```
Comment (sleepergoose) [inline frontend/src/app/modules/perflow-studio/perflow-studio-routing.module.ts]: Is a comment line necessary?

**Review by sleepergoose — CHANGES_REQUESTED** [review]

**Comment by proskurnicha** [comment]
Please fix conflicts and comments

**Review by proskurnicha — APPROVED** [review]

**Pull request merged**